src/transform/strategies/age_to_iso8601.rs

```diff
@@ -0,0 +1,241 @@
+///
+/// If an entry is already in ISO8601 duration format, it will be left unchanged.
+///
+/// If there are cell values which are neither ISO8601 durations nor integers
+/// an error will be returned.
+pub struct AgeToIso8601Strategy {
+    min_age: i32,
+    max_age: i32,
+}
+
+impl Default for AgeToIso8601Strategy {
+    fn default() -> Self {
+        AgeToIso8601Strategy {
+            min_age: 0,
+            max_age: 150,
+        }
+    }
+}
+
+impl AgeToIso8601Strategy {
+    pub fn new(min_age: i32, max_age: i32) -> Self {
+        AgeToIso8601Strategy { min_age, max_age }
+    }
+
+    fn is_valid_age(&self, age: i32) -> bool {
+        age >= self.min_age && age <= self.max_age
+    }
+}
+
+impl Strategy for AgeToIso8601Strategy {
+    fn is_valid(&self, tables: &[&mut ContextualizedDataFrame]) -> bool {
+        tables.iter().any(|table| {
+            !table
+                .filter_columns()
+                .where_header_context(Filter::Is(&Context::None))
+                .where_data_context_is_age()
+                .collect()
+                .is_empty()
+        })
+    }
+
+    fn internal_transform(
+        &self,
+        tables: &mut [&mut ContextualizedDataFrame],
+    ) -> Result<(), StrategyError> {
+        info!("Applying AgeToISO8601 strategy to data.");
+
+        let min_age = 0;
+        let max_age = 150;
+
+        let i32_to_iso8601: HashMap<i32, String> =
+            (min_age..=max_age).map(|n| (n, format!("P{n}Y"))).collect();
+
+        let mut error_info: HashSet<MappingErrorInfo> = HashSet::new();
+
+        for table in tables.iter_mut() {
+            let column_names: Vec<PlSmallStr> = table
+                .filter_columns()
+                .where_header_context(Filter::Is(&Context::None))
+                .where_data_context_is_age()
+                .collect()
+                .iter()
+                .map(|col| col.name())
+                .cloned()
+                .collect();
+
+            for col_name in column_names {
+                let col = table.data().column(&col_name)?;
+                let cast_col = col.cast(&DataType::String)?;
+                let mapped_column = cast_col.str()?.apply_mut(|cell_value| {
+                    if is_iso8601_duration(cell_value) {
+                        cell_value
+                    } else if let Ok(years) = cell_value.parse::<i32>()
+                        && self.is_valid_age(years)
+                    {
+                        i32_to_iso8601
+                            .get(&years)
+                            .expect("Age was too high or too low")
+                    } else if let Ok(years) = cell_value.parse::<f64>()
+                        && years.fract() == 0.0
+                        && self.is_valid_age(years as i32)
+                    {
+                        i32_to_iso8601
+                            .get(&(years as i32))
+                            .expect("Age was too high or too low")
```

Comment:
Might be better to first try to cast the value correctly and only then call `i32_to_iso8601`